src/librustc/session/config.rs

```diff
@@ -109,6 +109,7 @@ impl OutputType {
             OutputType::Bitcode => "bc",
             OutputType::Assembly => "s",
             OutputType::LlvmAssembly => "ll",
+            OutputType::Object if cfg!(target_os = "windows") => "obj",
             OutputType::Object => "o",
```

Comment:
@rkruppe I made a simple program that checks that. It didn't trigger a dead code warning.
@eddyb I agree. I'll see how it can be done.